subprojects/docs/src/snippets/testKit/automaticClasspathInjectionCustomTestSourceSet/groovy/build.gradle

```diff
@@ -4,25 +4,16 @@ plugins {
     id 'java-gradle-plugin'
 }
 
-sourceSets {
-    functionalTest {
-        groovy {
-            srcDir file('src/functionalTest/groovy')
-        }
-        resources {
-            srcDir file('src/functionalTest/resources')
-        }
-        compileClasspath += sourceSets.main.output + configurations.testRuntimeClasspath
-        runtimeClasspath += output + compileClasspath
-    }
-}
-
-task functionalTest(type: Test) {
+def functionalTest = sourceSets.create('functionalTest')
```

Comment:
Unless I missed something, this does not fully replace the above `functionalTest` configuration.
At least the modification of the `compileClasspath` is still needed.

Reply:
Yes that's true. But it is not needed for running a functional plugin test. Because you do not need the classes from the plugin in the test itself. I don't know why it was done like this before. Maybe this was needed for other reasons in older Gradle versions.

I would like to keep this minimal. It also matches how I did it for plugins I have been working on.

The test for this snippet also shows that this is enough.